src/indexer-helpers.ts

```
@@ -161,36 +194,57 @@ export async function loadIndex<K extends IndexKeyType, T extends DocFragment, C
+      id: r.id,
+    }));
+
+    if (query.since) {
+      const gen = clockChangesSince(crdt.blockstore, crdt.clock.head, query.since, query.sinceOptions || {}, logger);
+      const ids = await arrayFromAsyncIterable(gen)
+        .then((arr) => arr.map((a) => a.id))
+        .then((arr) => new Set(arr));
+      result = result.reduce((acc: ProllyIndexRow<K, R>[], row) => {
+        if (ids.has(row.id)) {
+          ids.delete(row.id);
+          return [...acc, row];
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)